fiat-roles/src/main/groovy/com/netflix/spinnaker/fiat/permissions/RedisPermissionsRepository.java

```diff
@@ -80,40 +83,40 @@
 
   @Override
   public RedisPermissionsRepository put(@NonNull UserPermission permission) {
-    try (Jedis jedis = jedisSource.getJedis()) {
-      for (ResourceType r : ResourceType.values()) {
-        Map<String, String> resourceValues = new HashMap<>();
+    Map<ResourceType, Map<String, String>> resourceTypeToRedisValue =
+        new HashMap<>(ResourceType.values().length);
 
-        Consumer<Resource> putInValuesMap = namedResource -> {
+    permission
+        .getAllResources()
+        .forEach(resource -> {
           try {
-            resourceValues.put(namedResource.getName(),
-                               objectMapper.writeValueAsString(namedResource));
+            resourceTypeToRedisValue
+                .computeIfAbsent(resource.getResourceType(), key -> new HashMap<>())
+                .put(resource.getName(), objectMapper.writeValueAsString(resource));
           } catch (JsonProcessingException jpe) {
             log.error("Serialization exception writing " + permission.getId() + " entry.", jpe);
           }
-        };
-
-        switch (r) {
-          case ACCOUNT:
-            permission.getAccounts().stream().forEach(putInValuesMap);
-            break;
-          case APPLICATION:
-            permission.getApplications().stream().forEach(putInValuesMap);
-            break;
-          case SERVICE_ACCOUNT:
-            permission.getServiceAccounts().stream().forEach(putInValuesMap);
-            break;
-        }
+        });
+
+    try (Jedis jedis = jedisSource.getJedis()) {
+      Pipeline pipeline1 = jedis.pipelined();
```

Comment:
Or at least comment and explain what is happening in each pipeline if they're doing different types of operations.